gamemode/modules/base/sv_interface.lua

```diff
@@ -67,6 +67,12 @@ DarkRP.retrievePlayerData = DarkRP.stub{
             description = "The function that receives the information.",
             type = "function",
             optional = false
+        },
+        {
+            name = "failure",
+            description = "The function that is called when the information cannot be retrieved.",
+            type = "function",
+            optional = false
```

Comment:
You calling it a micro optimisation makes as little sense to me as calling it a check. It's a callback.